packages/core/database/manager.go

```diff
@@ -27,7 +27,7 @@ type Manager struct {
 
 	openDBs         *cache.Cache[epoch.Index, *dbInstance]
 	bucketedBaseDir string
-	dbSizes         *shrinkingmap.ShrinkingMap[epoch.Index, int64]
+	dbSizes         *memstorage.Storage[epoch.Index, int64]
```

Comment:
This shrinkingmap is protected by the openDBsMutex. By removing the lock in the `PrunableStorageSize` is was made unsafe. So if we add the lock back in, then this can be reverted.